internal/reminder/reminder.go

```diff
@@ -144,19 +147,64 @@ func (r *reminder) sendReminders(ctx context.Context) []error {
 
 	logger.Info().Msgf("created repository batch of size: %d", len(repos))
 
-	// Update the reminder_last_sent for each repository to export as metrics
+	messages := make([]*message.Message, 0, len(repos))
+	errorSlice := make([]error, 0)
+
+	tx, err := r.store.BeginTransaction()
+	if err != nil {
+		return []error{err}
+	}
+
+	defer tx.Rollback()
+
+	qtx := r.store.GetQuerierWithTransaction(tx)
+
+	// TODO: Collect Metrics
+	// Potential metrics:
+	// - Gauge: Number of reminders in the current batch
+	// - UpDownCounter: Average reminders sent per batch
+	// - Histogram: reminder_last_sent time distribution
```

Comment:
How do we want to export the metrics? And some thoughts on adding these metrics?

Reply:
Hmm, I wonder what metrics would be useful as a minder operator.

About the histogram, I think a histogram that shows me how out of date (how much before a cutoff) a repo was when it was selected would be useful.

I think the metrics would be interesting to fine-tune the algorithm, did you see some other value in the metrics?

Since reminder is a separate service, then I guess implementation-wise it should just have its own metrics server that could be scraped separately.

Reply:
I'll add monitoring in a sep PR.

Reply:
What I was thinking of recording on each pass of collecting reminders would effectively be the following for each reminded row:

```
now() - reminder_last_sent
```

Which would _hopefully_ give a nice even distribution between `MinElapsed` and `MinElapsed + Interval` in normal operation.  Unfortunately, common values might be `MinElapsed = 24h` and `Interval = 5m`, which will lead to a histogram where everything lands in a very large bucket.  So, for practicality, I'd probably report this as:

```
now() - reminder_last_sent - MinElapsed
```

Which should give a range `0 .. Interval`.  If the algorithm starts to get behind or reminders "bunch", we'd see the tail of the histogram get bigger, and go beyond `Interval`.  Given the target of `Interval = 5m` (hypothetical, but seems reasonable), we'd probably want to measure in seconds with a starting bucket size of 10s, roughly-doubling buckets (or a custom 1..5..10 bucketer) and going out to at least 20 minutes, which would mean about 9 buckets (0, 10, 20, 40, 80, ... 1280).  I'm less concerned about the exact bucket sizing as being able to see whether most items are ending up within that `MinElapsed + Interval` window.

I'd store this metric in `UpdateReminderLastSentForRepositories`, so that the rest of the loop didn't need to care, and it was next to the handling of `reminder_last_sent`.